Paintroid/src/main/java/org/catrobat/paintroid/tools/helper/JavaCropAlgorithm.kt

```diff
@@ -0,0 +1,111 @@
+/*
+ * Paintroid: An image manipulation application for Android.
+ * Copyright (C) 2010-2021 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.paintroid.tools.helper
+
+import android.graphics.Bitmap
+import android.graphics.Color
+import android.graphics.Rect
+import android.util.Log
+
+private const val TAG = "cropAlgorithmSnail"
+
+class JavaCropAlgorithm : CropAlgorithm {
+    private fun containsOpaquePixel(
+        pixels: Array<IntArray>,
+        fromX: Int,
+        fromY: Int,
+        toX: Int,
+        toY: Int
+    ): Boolean {
+        for (y in fromY..toY) {
+            for (x in fromX..toX) {
+                if (pixels[y][x] != Color.TRANSPARENT) {
+                    return true
+                }
+            }
+        }
+        return false
+    }
```

Comment:
```suggestion
    private fun lineContainsOpaquePixel(bounds: Rect, pixels: Array<IntArray>, y) {
       ...
    }
    private fun columnContainsOpaquePixel(bounds: Rect, pixels: Array<IntArray>, x) {
       ...
    }
```
https://kotlinlang.org/docs/collection-elements.html#retrieve-by-condition